src/init/processAnnotations.ts

```diff
@@ -0,0 +1,62 @@
+import store from 'store';
+import { mousePosition } from 'common/dom';
+import _isEqual from 'lodash/isEqual';
+import { selectAnnotations } from '../store/api/selectors';
+import { findUniqueTextInDOMAsRange } from '../utils/annotations';
+import { locateAnnotations } from '../store/annotations/actions';
+import { LocatedAnnotation } from '../store/annotations/types';
+import { AnnotationAPIModel } from '../api/annotations';
+
+let instance;
+
+function init() {
+  // subscribe to store changes and return unsubscribe fn
+  const unsubscribe = store.subscribe(processAnnotations);
```

Comment:
Hmm, listening to every change of store(subscribe) and then for any change, locate all annotations. I know it was worse than it is now, but sooner or later I would aim for reducers instead of subscriptions (attached to three actions API_CREATED, API_READ, API_UPDATED) and components tree instead of "manual" drawing of highlights. Then react+redux can optimize re-rendering and the whole flow will be more under control.

Reply:
As we agreed at the meeting, this should be eventually done. I opened #209